### PR TITLE
Military time validator v1

### PR DESCRIPTION
**Pull request by rochaalexandre**
## Problem Description
 Description: Write a function (or a stateless class) capable of validating whether a string time range is a valid military time range or not.

Here are some string examples:
- "01:12 - 14:32" (yes)
- "25:00 - 12:23" (no)
- "22:00 - 23:12" (yes)

### CODE REVIEW

File: 1_The_Physical/Level_1_Stateless_Guess_Point/2_Manage_Complexity/3_1_Military_Time_Validator/src/index.ts [hunk 1]
```diff
@@ -1,7 +1,11 @@
 export class TimeValidator {
   private static readonly EARLIEST_HOUR: String = "0001";
   private static readonly LASTEST_HOUR: string = "2359";
+  private static readonly RANGE_PATTERN = /(\d{2}:\d{2})\s-\s(\d{2}:\d{2})/;
   static validateRange(timeRange: string): boolean {
+    if (!this.RANGE_PATTERN.test(timeRange)) {
```
Comment (realraif) [inline 1_The_Physical/Level_1_Stateless_Guess_Point/2_Manage_Complexity/3_1_Military_Time_Validator/src/index.ts]: Is it really the lowest TPP option?

File: 1_The_Physical/Level_1_Stateless_Guess_Point/2_Manage_Complexity/3_1_Military_Time_Validator/src/index.spec.ts [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+    ('knows that "%s" is valid military time range ', (param) => {
+        expect(TimeValidator.validateRange(param)).toBeTruthy()
+    });
+
+    it.each(["25:00 - 12:23", "15:00 - 14:32", "00:01 - 00:00", "22:00 - 21:12", "15:30 - 25:30"])
+    ('knows that "%s" is not a valid military time range ', (param) => {
+        expect(TimeValidator.validateRange(param)).toBeFalsy()
+    });
+
+    it.each(["25:00 - 23", "aa:aa - 0", ""])
```
Comment (realraif) [inline 1_The_Physical/Level_1_Stateless_Guess_Point/2_Manage_Complexity/3_1_Military_Time_Validator/src/index.spec.ts:15]: You're using really good validation examples that contribute to the TDD process.

It's good practice to represent the validation errors you're testing for in the test name
(like "knows that it's not an empty string" or "knows that the hour clock is up to 24")

**Review by realraif — COMMENTED** [review]
Good job with the work process and final result!

File: 1_The_Physical/Level_1_Stateless_Guess_Point/2_Manage_Complexity/3_1_Military_Time_Validator/src/index.ts (hunk 1 above)
Comment (rochaalexandre) [reply 1_The_Physical/Level_1_Stateless_Guess_Point/2_Manage_Complexity/3_1_Military_Time_Validator/src/index.ts]: To be honest I don't think so, but it was the simplest one for me at that time

**Pull request merged**